src/Pos/Util/UserSecret.hs

```diff
@@ -134,7 +184,19 @@ writeUserSecretRelease u
 
 -- | Helper for writing secret to file
 writeRaw :: UserSecret -> IO ()
-writeRaw u = BSL.writeFile (u ^. usPath) $ encode u
+writeRaw u = do
+    let path = u ^. usPath
+    -- On POSIX platforms, openTempFile guarantees that the file
+    -- will be created with mode 600.
```

Comment:
@vincenthz you're right about `openBinaryTempFile`, I'll fix that. Also, please see the [docs](https://hackage.haskell.org/package/base-4.9.1.0/docs/System-IO.html#v:openTempFile) for `openTempFile`:

> The file is created with permissions such that only the current user can read/write it.